pintc/tests/tests.rs

```diff
@@ -53,8 +53,11 @@ fn run_tests(sub_dir: &str) -> anyhow::Result<()> {
                 // Type check the parsed intent.
                 type_check(pred, &test_data, &mut failed_tests, &path))
             .and_then(|pred|
-                // Flatten the parsed intent check the result.
-                flatten_and_check(pred, &test_data, &mut failed_tests, &path));
+                // Flatten the parsed intent and check the result.
+                flatten_and_check(pred, &test_data, &mut failed_tests, &path))
+            .and_then(|pred|
+                // Optimise the flattened intent
+                optimise(pred, &test_data));
```

Comment:
I think we discussed adding a separate step and a separate section for optimized code in the tests. This doesn't seem to be doing that.

Reply:
It is separated in `optimise()` based on whether or not the test has `optimise >>>`. If it's not present we skip the step.

```rust
fn optimise(pred: Contract, test_data: &TestData) -> Option<Contract> {
    let optimised = pred.optimise();
    if let Some(expected_optimised_str) = &test_data.optimised {
        similar_asserts::assert_eq!(expected_optimised_str.trim(), format!("{optimised}").trim());
        Some(optimised)
    } else {
        None
    }
}
```